File: algorithms/javascript/bubble_sort.js
function BubbleSort(arr) {
    const sortedArray = Array.from(arr);
    let swap;
    do {
        swap = false;
        for (let i = 1; i < sortedArray.length; ++i) {
            if (sortedArray[i - 1] > sortedArray[i]) {
                [sortedArray[i], sortedArray[i - 1]] = [sortedArray[i - 1], sortedArray[i]];
                swap = true;
            }
        }
    } while (swap)
    return sortedArray;
}

File: algorithms/cpp/bubble_sort.cpp
#include <iostream>
using namespace std;

void bubSort(int arr[], int n)
{
    for (int i = 0; i < n-1; i++)
    for (int j = 0; j < n-i-1; j++)
        if (arr[j] > arr[j+1])
            swap(arr[j], arr[j+1]);
}

int main()
{
    int n;
    cin>>n;

    int*arr=new int[n];

    for(int i=0;i<n;i++)
        cin>>arr[i];

    bubSort(arr,n);

    for(int i=0;i<n;i++)
        cout<<arr[i]<<" ";
    
    return 0;
}

